src/ape_ethereum/provider.py

```diff
@@ -624,6 +624,25 @@ def get_receipt(
         )
         hex_hash = HexBytes(txn_hash)
 
+        txn = {}
+        if transaction := kwargs.get("transaction"):
+            # perf: If called `send_transaction()`, we should already have the data!
+            txn = (
+                transaction
+                if isinstance(transaction, dict)
+                else transaction.model_dump(by_alias=True, mode="json")
+            )
+
+        if kwargs.get("private"):
+            # Bail before confirmation because it won't be on chain yet.
+            data = {
+                "required_confirmations": 0,
```

Comment:
**problem**: this still needs to be set to the correct value so that when the user _does_ call `wait()`, it will wait for the correct amount of confs.

**Note**: I do see why you are doing this waiting for the nonce bump but I added a different  comment below re: that